File: my-ai-dashboard/types/store.ts
export type StoreItem = {
  id: number
  name: string
  city: string | null
  is_active: boolean
}

export type StoreListData = {
  items: StoreItem[]
  total: number
  page: number
  page_size: number
  sort_by: string
  sort_order: string
}

File: backend/app/schemas/store.py
from typing import Optional

from pydantic import BaseModel


class StoreListItem(BaseModel):
    id: int
    name: str
    city: Optional[str] = None
    is_active: bool


class StoreListData(BaseModel):
    items: list[StoreListItem]
    total: int
    page: int
    page_size: int
    sort_by: str
    sort_order: str


class StoreListResponse(BaseModel):
    code: int
    message: str
    data: StoreListData
    request_id: Optional[str] = None
    timestamp: str
